website/docs/r/ecs_disk.html.markdown

```diff
@@ -0,0 +1,92 @@
+---
+subcategory: "ECS"
+layout: "alicloud"
+page_title: "Alicloud: alicloud_ecs_disk"
+sidebar_current: "docs-alicloud-resource-ecs-disk"
+description: |-
+  Provides a Alicloud ECS Disk resource.
+---
+
+# alicloud\_ecs\_disk
+
+Provides a ECS Disk resource.
+
+For information about ECS Disk and how to use it, see [What is Disk](https://www.alibabacloud.com/help/en/doc-detail/25513.htm).
+
+-> **NOTE:** Available in v1.122.0+.
+
+## Example Usage
+
+Basic Usage
+
+```terraform
+resource "alicloud_ecs_disk" "example" {
+  availability_zone = "cn-beijing-b"
+  name              = "tf-test"
+  description       = "Hello ecs disk."
+  category          = "cloud_efficiency"
+  size              = "30"
+  encrypted         = true
+  kms_key_id        = "2a6767f0-a16c-4679-a60f-13bf*****"
+  tags = {
+    Name = "TerraformTest"
+  }
+}
+
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `availability_zone` - (Optional, ForceNew) ID of the free zone to which the disk belongs.
+* `category` - (Optional) Category of the disk. Valid values are `cloud`, `cloud_efficiency`, `cloud_ssd`, `cloud_essd`. Default is `cloud_efficiency`.
+* `delete_auto_snapshot` - (Optional) Indicates whether the automatic snapshot is deleted when the disk is released. Default value: `false`.
+* `delete_with_instance` - (Optional) Indicates whether the disk is released together with the instance. Default value: `false`.
+* `description` - (Optional) Description of the disk. This description can have a string of 2 to 256 characters, It cannot begin with http:// or https://. Default value is null.
+* `disk_name` - (Optional, Computed) Name of the ECS disk. This name can have a string of 2 to 128 characters, must contain only alphanumeric characters or hyphens, such as "-",".","_", and must not begin or end with a hyphen, and must not begin with http:// or https://. Default value is null.
```

Comment:
such as "-", ".", "_", and must not begin or end with a hyphen, and must not begin with \`http://\` or \`https://\`. Default value is \`null\`.